web/src/ui_init.js

```diff
@@ -205,6 +205,36 @@ function initialize_right_sidebar() {
             }
         }
     });
+
+    $("#buddy-list-users-matching-view-container").on(
+        "click",
+        ".buddy-list-subsection-header",
+        () => {
+            $("#buddy-list-users-matching-view").toggleClass("collapsed");
+            const is_collapsed = $("#buddy-list-users-matching-view").hasClass("collapsed");
+            $("#buddy-list-users-matching-view-container .toggle-narrow-users").toggleClass(
+                "fa-caret-down",
+                !is_collapsed,
+            );
+            $("#buddy-list-users-matching-view-container .toggle-narrow-users").toggleClass(
+                "fa-caret-left",
+                is_collapsed,
+            );
+        },
+    );
+
+    $("#buddy-list-other-users-container").on("click", ".buddy-list-subsection-header", () => {
+        $("#buddy-list-other-users").toggleClass("collapsed");
+        const is_collapsed = $("#buddy-list-other-users").hasClass("collapsed");
+        $("#buddy-list-other-users-container .toggle-other-users").toggleClass(
+            "fa-caret-down",
+            !is_collapsed,
+        );
+        $("#buddy-list-other-users-container .toggle-other-users").toggleClass(
+            "fa-caret-left",
+            is_collapsed,
+        );
+    });
```

Comment:
I was considering trying to set this up more generally instead of mostly copying the same code, but left it as this for now since I think it's more readable.

Reply:
Yeah, this seems fine. It would be nice to move `initialize_right_sidebar` and `initialize_left_sidebar` into `sidebar_ui.js`. Let's plan to move those functions; could be in a prep PR if that's easy.

Reply:
#27484